Source/WebCore/dom/MouseEvent.idl

```diff
@@ -54,8 +54,8 @@
 
     readonly attribute long offsetX;
     readonly attribute long offsetY;
-    readonly attribute long x;
-    readonly attribute long y;
+    [ImplementedAs=clientX] readonly attribute long x;
+    [ImplementedAs=clientY] readonly attribute long y;
```

Comment:
Why did you decide to do this rather than just leaving `x` and `y` functions in the MouseEvent class? Some day maybe we could rename `clientX` and `clientY` in the C++ code?

Perhaps it’s good to retain the “client” names to add clarity?

Reply:
Some more of my thinking: To me these “client” names are not unambiguous and don’t read like names we would select for clarity. On the other hand, since there is ambiguity between coordinate systems, I could indeed see us using names not as short as just `x` and `y`. A function that returns a point is what we would create internally; separate x and y functions would just be at the interface edge for compatibility with DOM standards.

Reply:
Admittedly I did not think about this a whole lot, but standards-wise `x` and `y` came later and internally we have `m_clientLocation` (used by these) and `m_pageLocation` (used by some others).